pkg/apis/core/validation/validation.go

```diff
@@ -1898,7 +1901,16 @@ func ValidateEnv(vars []core.EnvVar, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
-var validFieldPathExpressionsEnv = sets.NewString("metadata.name", "metadata.namespace", "metadata.uid", "spec.nodeName", "spec.serviceAccountName", "status.hostIP", "status.podIP")
+var validEnvDownwardAPIFieldPathExpressions = sets.NewString(
+	"metadata.annotations[]", // represents "metadata.annotations" with an arbitary subscript
+	"metadata.labels[]",      // represents "metadata.labels" with an arbitary subscript
```

Comment:
same question here.

Reply:
This is used only in Downward API; the original name was misleading.